Fixed #32004 -- Clarified docs for when request.FILES is set. (#13425)
django/django: docs/topics/http/file-uploads.txt

be accessible as ``request.FILES['file']``.

Note that :attr:`request.FILES <django.http.HttpRequest.FILES>` will only

> contain data from form fields if the request method was ``POST`` and
> the ``<form>`` that posted the request has the attribute
> ``enctype="multipart/form-data"`` and contains at least one
> ``<intput type="file">`` field.
> Or more technically if request method was ``POST``, the
> ``Content-Type`` general header was ``multipart/form-data`` and
> the ``Content-Disposition`` subpart header contains the filename parameter
> (default behaviour in browsers for this ``enctype``).
> Otherwise, ``request.FILES`` will be empty.

Reviewer: I think I might trim it down to maybe just this: 

```suggestion
contain data from form fields if the request method was ``POST``, at least one
file field was actually posted, and the ``<form>`` that posted the request has
the attribute ``enctype="multipart/form-data"``.
```

I think, in this context, the whole _more technically_ bit is just too much.

Author: I'm fine with it, it's definitely more precise than the current version.
Thanks for your help and time!

Cheers

be accessible as ``request.FILES['file']``.

Note that :attr:`request.FILES <django.http.HttpRequest.FILES>` will only
contain data from form fields if the request method was ``POST``, at least one

> file field was actually posted, and the ``<form>`` that posted the request has

Reviewer 2: I notice Oxford commas used quite a bit in the docs. Is there a reason for this? Personally, I'm not sure they aid readability in scenarios such as this. 

An example where I think they are helpful: 'I went shopping at marks and spencers, and tesco'. 

I guess what I'm really asking is this a conscious 'house style'? If so I'll bear in mind for the future. :-)

Reviewer: Yep, always use them. (Sorry 🙂)

Reviewer 3: We :heart: them :wink:

Reviewer 2: Thanks both :-1
